src/koheesio/pandas/categorical_encoding.py

```diff
@@ -0,0 +1,57 @@
+from sklearn.preprocessing import OneHotEncoder
+import pandas as pd
+from pydantic import BaseModel
+from typing import List, Optional
+from koheesio.steps import Step
+
+from typing import List, Dict
+import pandas as pd
+from pydantic import BaseModel
+
+class PandasCategoricalEncoding(BaseModel):
+    """
+    Encodes categorical columns using one-hot encoding or ordinal encoding.
+
+    Attributes
+    ----------
+    columns : List[str]
+        List of categorical columns to encode.
+    encoding_type : str, optional
+        Type of encoding, either "one-hot" or "ordinal" (default is "one-hot").
+    drop_first : bool, optional
+        Whether to drop the first dummy column for one-hot encoding (default is True).
+    ordinal_mapping : dict, optional
+        A dictionary mapping categorical values to integers for ordinal encoding.
+    """
+
+    columns: List[str]
+    encoding_type: str = "one-hot"
```

Comment:
change the type to `Literal["one-hot", "ordinal"]`, that way you don't need the extra check you put in the `__init__` method